File: frontend/src/domain/products/domain/models/ProductSupplier.js
class ProductSupplier {
    constructor({ id, productId, supplierId, price }) {
      this.id = id || null;
      this.productId = productId || null;
      this.supplierId = supplierId || null;
      this.price = price || 0.0;
    }
  }
  
  export default ProductSupplier;
  

File: backend/supplier_service/src/main/java/ucv/app_inventory/supplier_service/domain/model/ProductSupplier.java
package ucv.app_inventory.supplier_service.domain.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "product_supplier")
public class ProductSupplier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(name = "product_id")
    private Long productId;

    @NotNull
    @Column(name = "supplier_id")
    private Long supplierId;

    @NotNull
    private Double price;
}
